src/NBitcoin/IConsensus.cs

```diff
@@ -104,9 +104,16 @@ public interface IConsensus
         /// </summary>
         ConsensusFactory ConsensusFactory { get; }
 
-        /// <summary>
-        /// Rules specific to the given network.
-        /// </summary>
-        ICollection<IConsensusRule> Rules { get; set; }
+        /// <summary>Group of rules that are used during block's header validation specific to the given network.</summary>
+        List<ISyncBaseConsensusRule> HeaderValidationRules { get; set; }
+
+        /// <summary>Group of rules that are used during block integrity validation specific to the given network.</summary>
+        List<ISyncBaseConsensusRule> IntegrityValidationRules { get; set; }
+
+        /// <summary>Group of rules that are used during partial block validation specific to the given network.</summary>
+        List<IAsyncBaseConsensusRule> PartialValidationRules { get; set; }
+
+        /// <summary>Group of rules that are used during full validation (connection of a new block) specific to the given network.</summary>
+        List<IAsyncBaseConsensusRule> FullValidationRules { get; set; }
     }
```

Comment:
For any types in the comments  - like Network - use <see>

Reply:
here I'm not referencing network class. I mean different network in general- like stratis, bitcoin etc